azure-iot-device/azure/iot/device/iothub/transport/pipeline_stages_iothub.py

```diff
@@ -37,7 +37,7 @@ def _run_op(self, op):
             self.run_ops_serial(
                 pipeline_ops_iothub.SetAuthProviderArgs(
                     device_id=auth_provider.device_id,
-                    module_id=auth_provider.module_id,
+                    module_id=getattr(auth_provider, "module_id", None),
```

Comment:
>module_id=getattr(auth_provider, "module_id", None), [](start = 20, length = 52)

This is odd, why is this default to `None` necessary? The value for `module_id` alerady defaults to `None` in the auth provider class, why do we need it here as well?

Reply:
defensive programming.  I didn't consult the providers and just considered the idea that module_id isn't required.  

---
In reply to: [287536915](https://github.com/Azure/azure-iot-sdk-python-preview/pull/94#discussion_r287536915) [](ancestors = 287536915)